src/main/java/com/google/sps/servlets/EnterQueue.java

```diff
@@ -61,15 +73,59 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
             Key classKey = KeyFactory.stringToKey(classCode);
             Entity classEntity = datastore.get(txn, classKey);
 
-            ArrayList<String> updatedQueue = (ArrayList) classEntity.getProperty("studentQueue");
+            // Get date in mm/dd/yyyy format
+            DateTimeFormatter FOMATTER = DateTimeFormatter.ofPattern("MM/dd/yyyy");
+            LocalDate localDate = LocalDate.now(clock);
+            String currDate = FOMATTER.format(localDate);
+
+            Filter classVisitFilter =
+                new FilterPredicate("classCode", FilterOperator.EQUAL, classCode);
+
+            Filter dateVisitFilter = new FilterPredicate("date", FilterOperator.EQUAL, currDate);
+
+            CompositeFilter visitFilter =
+                CompositeFilterOperator.and(dateVisitFilter, classVisitFilter);
+
+            int retries2 = 10;
```

Comment:
If `retries` is never used can this be renamed to `retries`?

Reply:
retries2 is used in the inner transaction, retries is used in the outer transaction

Reply:
I'm not a big fan of appending numbers to the end of vars, maybe retriesInner to represent the inner transaction